internal/engine/upper_test.go

```diff
@@ -1023,6 +1023,44 @@ func TestPodUnexpectedContainerStartsImageBuild(t *testing.T) {
 	f.waitForCompletedBuildCount(1)
 }
 
+func TestPodUnexpectedContainerStartsImageBuildOutOfOrderEvents(t *testing.T) {
+	f := newTestFixture(t)
+	defer f.TearDown()
+	f.bc.DisableForTesting()
+
+	mount := model.Mount{LocalPath: "/go", ContainerPath: "/go"}
+	name := model.ManifestName("foobar")
+	manifest := f.newManifest(name.String(), []model.Mount{mount})
+
+	f.Start([]model.Manifest{manifest}, true)
+
+	// Start and end a fake build to set manifestState.ExpectedContainerId
```

Comment:
put some paragraph breaks in this code so i can tell where the setup code ends and the actual testing starts?

Reply:
done